### PR TITLE
Unify analysis with resolve preview, add version, add docs

### LINKED ISSUE #33: Resolution project from current directory is broken
When adding Argu, I made a regression.

We are forced to pass the project path.

Executing this command `dotnet ../../../../Zaid-Ajaj/Femto/bin/Debug/netcoreapp2.2/Femto.dll` print the manual instead of running Femto using the current directory.

### LINKED ISSUE #34: Should we allow passing a directory instead of the full path to the fsproj?
`dotnet` command allow the user to path a directory `dotnet restore src` and will find the `project` in the direct directory.

It only complains if there are several `project` files in the directory. Should we mimic this behaviour?

### LINKED ISSUE #38: Add --version
Adding `--version` to the input arguments.

### LINKED ISSUE #40: Unify resolve preview and project analysis logs
By @MangelMaxime [in this comment here](https://github.com/Zaid-Ajaj/Femto/pull/39#issuecomment-506288114)

> I asked my self this question in the past and yes they look similar but have small differences.
> --preview will fail is not package.json has been found.
> While --validate can be used even if no package.json is present.
> And --preview show what actions will be applied to the project. It will tell you that it will uninstall that version of the library and install this one.
> However, I do think that running femto without arguments is a duplicate of one of the feature we have.

> I think --preview and running without args should be the same and output the "nicely formatted" logs.

That's what I am thinking of too, not providing any arguments (except project file or path) should be an alias for `--preview` and `--preview` logs can be more nicely formatted

### PR DESCRIPTION
**Pull request by Zaid-Ajaj**
Fixes #40 
Fixes #38 
Fixes #34 
Fixes #33 

@MangelMaxime I will merge this right away and publish a new version

### CODE REVIEW

**Comment by Zaid-Ajaj** [comment]
Published as version `0.4.0` 🚀:tada: :tada: @MangelMaxime  can you please take it for a spin?

File: Program.fs [hunk 1]
```diff
@@ -665,10 +665,172 @@ let private validateProject (library : LibraryWithNpmDeps) =
...
+            | [ ResolveAction.Install(_, _, version, range) ] ->
+                if range.Contains "&&"
+                then logger.Information("  | -- Required version constraint from multiple projects [{Range}]", package, range)
+                let installationCommand =
+                    nodeCmd
+                      (sprintf "npm install %s@%s --save" package version)
+                      (sprintf "yarn add %s@%s" package version)
+                logger.Information("  | -- Resolve manually using '{Command}'", installationCommand)
+
+            | [ ResolveAction.UninstallDev(_); ResolveAction.Install(_, _, version, range) ] ->
```
Comment (MangelMaxime) [inline Program.fs]: Are we sure that `UninstallDev` and `Install` are always next to each other for the same library?

File: Program.fs [hunk 2]
```diff
@@ -819,42 +963,48 @@ let rec private runner (args : FemtoArgs) =
...
+        | Resolve :: rest ->
             apply rest { res with Resolve = true }
 
-        | Preview::rest ->
-            apply rest { res with ResolvePreview = true }
+        | Version :: rest ->
+            apply rest { res with DiplayVersion = true }
 
-        | [ ] ->
+        | _ ->
```
Comment (MangelMaxime) [inline Program.fs]: I am not a fan of using discard case in DUs. 

The compiler can't help us anymore when adding a new case. Is there another case to support that `[]` here?

**Comment by MangelMaxime** [comment]
By looking at the code it seems good to me. 

I left some comment. 

I will give it a try during the weekend or at the beginning of next week when publishing Bulma with Femto metadata now that `validate` should work for it.

File: Program.fs (hunk 1 above)
Comment (Zaid-Ajaj) [reply Program.fs]: Yes, that's how [I added them](https://github.com/Zaid-Ajaj/Femto/blob/master/Program.fs#L272-L275)

File: Program.fs (hunk 2 above)
Comment (Zaid-Ajaj) [reply Program.fs]: Yeah you are right, I will add it back

**Comment by Zaid-Ajaj** [comment]
Thanks for the review, now that I read my code again, I realized I missed a case 
```
[ ResolveAction.InstallDev( _ ) ] -> 
```
in the project analysis. I will fix it right away

**Pull request merged**